src/Microsoft.AspNetCore.JsonPatch/Helpers/ConversionResultProvider.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Newtonsoft.Json;
+
+namespace Microsoft.AspNetCore.JsonPatch.Helpers
+{
+    internal static class ConversionResultProvider
```

Comment:
`Helpers` is the first clue that something is wrong here. 

1. We should find a better way to do this conversion - for instance this should use `JsonContract`/`JsonConverter` and not do serialization.
2. The usability of this is wrong, this should be a `bool TryConvert(Type, object, out object)`